src/i2c.rs

```diff
@@ -2,16 +2,24 @@
 
 use core::iter::TrustedLen;
 use core::marker::PhantomData;
+use core::ops::Deref;
 use embedded_hal;
 use stm32f7::stm32f7x6::{
-    i2c1::{self, RegisterBlock},
+    i2c1,
     RCC,
+    self as device,
 };
 
-// TODO use &mut when svd2rust API has changed (modification should require &mut)
-//pub struct I2C<'a>(&'a mut RegisterBlock);
+pub trait I2cTrait: Deref<Target = i2c1::RegisterBlock> {
```

Comment:
neat trick

Reply:
This would be a perfect place for a [trait aliases](https://github.com/rust-lang/rust/issues/41517), but they're still unstable.